test/test_specialize.py

```diff
@@ -326,6 +326,102 @@ def foo(x: torch.Tensor, bitshift: tuple[int, int]) -> torch.Tensor:
         self.assertIn("65536", code)
         self.assertExpectedJournal(code)
 
+    def test_specialize_stride_basic(self):
+        """Test that hl.specialize works with tensor strides."""
+
+        @helion.kernel(config=helion.Config(block_sizes=[32, 32]))
+        def fn(x: torch.Tensor) -> torch.Tensor:
```

Comment:
i don't think this is quite demonstrating what i'd like because it has static_shapes on in the background. i want the generated triton code to be able to specialize on the stride even without static_shapes

Reply:
yup sorry forgot to update it to `static_shapes=False` - done and the tests still pass

Reply:
see how this now still passes all strides to the triton kernel?

Reply:
this is a general thing i've noticed, i always assumed that when i specialize it replaces out uses of the specialized value automatically, but doesn't seem so (same with unifying via assert)

Reply:
sg I filed https://github.com/pytorch/helion/issues/1216 to track this feature.

also added a unit test `test_specialize_size_becomes_static` in this PR to assert that after shape specialization, the triton kernel doesn't take shape var as argument.